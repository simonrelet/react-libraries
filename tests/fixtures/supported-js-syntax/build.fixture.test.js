'use strict'

const fixture = require('../../setupTests')

describe(`${fixture.fixtureName} build`, () => {
  it('should not fail', async () => {
    const result = await fixture.runScript('build')
    expect(result).toMatchSnapshot()
  })

  it('should generate all files', async () => {
    expect(await fixture.listFolderContent('build')).toEqual([
      'cjs/async.js',
      'cjs/ClassComponent.js',
      'cjs/DefaultExport.js',
      'cjs/DynamicImport/DynamicImport.js',
      'cjs/DynamicImport/LazyComponent.js',
      'cjs/FunctionComponent.js',
      'cjs/nullishCoalescing.js',
      'cjs/optionalChaining.js',
      'es/async.js',
      'es/ClassComponent.js',
      'es/DefaultExport.js',
      'es/DynamicImport/DynamicImport.js',
      'es/DynamicImport/LazyComponent.js',
      'es/FunctionComponent.js',
      'es/nullishCoalescing.js',
      'es/optionalChaining.js',
    ])
  })
})
