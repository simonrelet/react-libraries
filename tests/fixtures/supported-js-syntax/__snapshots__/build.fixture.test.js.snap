// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`supported-js-syntax build should not fail 1`] = `
Exit code: 0

--- Output ---
$ react-libraries build
CommonJS build successful.
ES modules build successful.
--------------
`;
